app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,203 @@
+        for (item in Gender.valuesAsStringRes()) {
+            sexValues += getString(item)
+        }
+        sexValues += getString(R.string.profile_spinner_prompt)
+        val sexAdapter = HintAdapter(this.activity, android.R.layout.simple_list_item_1, sexValues)
+        sexAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        sexSpinner.adapter = sexAdapter
+        sexSpinner.setSelection(sexAdapter.count)
+
+        // Set up birthday date picker
+        val dateText = root.findViewById<EditText>(R.id.birthday_edit_text)
+        val cal = Calendar.getInstance()
+        setupDatePicker(dateText, cal)
+
+        // Set up text input view hints
+        val feetContainer = root.findViewById<TextInputLayout>(R.id.height_feet_container)
+        val inchesContainer = root.findViewById<TextInputLayout>(R.id.height_inches_container)
+        val weightContainer = root.findViewById<TextInputLayout>(R.id.weight_container)
+
+        feetText.onFocusChangeListener = TextInputViewHintListener(feetContainer,
+                getString(R.string.profile_height_prompt), getString(R.string.profile_height_feet_units))
+        inchesText.onFocusChangeListener = TextInputViewHintListener(inchesContainer,
+                getString(R.string.profile_height_inches_prompt), getString(R.string.profile_height_inches_units))
+        weightText.onFocusChangeListener = TextInputViewHintListener(weightContainer,
+                getString(R.string.profile_weight_prompt), getString(R.string.profile_weight_lbs_units))
+
+        // Enable button iff all fields are empty
+        val editTextList = listOf(bdayPicker, feetText, inchesText, weightText)
+        sexSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                submitButton.isEnabled = areAllFieldsEnabled()
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {
+                ;
+            }
+        }
+
+        val watcher = ButtonTextWatcher()
+        for (editText in editTextList) {
+            editText.addTextChangedListener(watcher)
+        }
+
+        // Set button on click behavior
+        submitButton.setOnClickListener{
+            val sex = sexSpinner.selectedItemPosition
+            val age = cal.timeInMillis
+            val feet = feetText.text.toString()
+            val inches = inchesText.text.toString()
+            val pounds = weightText.text.toString()
+
+            val builder = AlertDialog.Builder(this.activity)
+            builder.setPositiveButton("OK",  {_, _ ->
+                ;
+            })
+            val user = UserProfile(Gender.values()[sex], age,
+                    heightToInches(feet.toInt(), inches.toInt()), pounds.toInt())
+            if (age.toInt() < MIN_AGE) {
+                builder.setMessage(R.string.error_profile_underage)
```

Comment:
Entering a birthdate of 2/17/1996 gives an error that I need to be at least 13 years old to use the application. I need an adult, Alex.